www/include/views/graphs/exportData/ExportCSVServiceData.php

```diff
@@ -138,20 +138,19 @@ function get_error($str)
     while ($indexData = $stmt->fetch(\PDO::FETCH_ASSOC)) {
         $listMetric[$indexData['metric_id']] = $indexData['metric_name'];
         $listEmptyMetric[$indexData['metric_id']] = '';
-        if ($start !== false && $end !== false) {
-            $stmt2 = $pearDBO->prepare(
-                "SELECT ctime, `value` FROM data_bin WHERE id_metric = :metricId " .
-                "AND ctime >= :start AND ctime < :end"
-            );
-            $stmt2->bindValue(':start', $start, \PDO::PARAM_INT);
-            $stmt2->bindValue(':end', $end, \PDO::PARAM_INT);
-            $stmt2->bindValue(':metricId', $indexData['metric_id'], \PDO::PARAM_INT);
-            $stmt2->execute();
-            while ($data = $stmt2->fetch(\PDO::FETCH_ASSOC)) {
-                $datas[$data["ctime"]][$indexData["metric_id"]] = $data["value"];
-            }
-        } else {
-            die('Start and end time were not provided');
+        if ($start === false && $end === false) {
```

Comment:
as start and end values won't change while looping, you need to check them before the line 125 to avoid to use resources for nothing